rep-0147.rst

```diff
@@ -0,0 +1,365 @@
+    C[Controller]
+    L[ ]
+    style L fill:#fff,stroke:#ff0,stroke-width:0px;
+    L --> CS
+    L --> GCS
+    L --> OS
+    subgraph Generic Planning Framework
+    CM[Collision Mapping]
+    MSC[Minimum Snap Controller]
+    CM ---|Shared State|MSC
+    end
+    CS -->|Obstacle Observations|CM
+    OS -->|Odometry|MSC
+    OS -->|Odometry|CM
+    OS -->|Odometry|C
+    GCS -->|Goals|MSC
+    MSC -->|Trajectory|C
+    %% Links to L 
+    linkStyle 0 fill:#fff,stroke:#ff0,stroke-width:0px;
+    linkStyle 1 fill:#fff,stroke:#ff0,stroke-width:0px;
+    linkStyle 2 fill:#fff,stroke:#ff0,stroke-width:0px;
+  </div>
+
+
+Coordinate Frames
+=================
+
+It is expected that users of this interface will also follow the coordinate frame conventions layed out in REP 105 [#rep105]_.
+
+Proposed Interfaces
+===================
+
+Control Interfaces
+------------------
+
+Trajectory Control Interface
+,,,,,,,,,,,,,,,,,,,,,,,,,,,,
+
+For sending trajectories in 3D space there already exists a message used for free space planning.
+This message used is ``trajectory_msgs/MultiDOFJointTrajectory`` [#multidoftraj]_ and is actively used by planning frameworks already.
+
+This message is not the most straight forward to parse as it has a lot of optional fields which increases the complexity and requires consistent labelling and filling.
+The complexity of the message parsing is traded off against supporting a relatively compact representations.
+
+* Datatype: ``trajectory_msgs/MultiDOFJointTrajectory`` [#multidoftraj]_ 
+* Topic: ``command_trajectory``
+
+Pose Interface
+,,,,,,,,,,,,,,
+
+The message to send a goal pose is ``geometry_msgs/PoseStamped`` [#posestamped]_.
+This is a very highly used standard message and contains precisely the information needed.
+
+* Datatype: ``geometry_msgs/PoseStamped`` [#posestamped]_
+* Topic: ``command_pose``
+
+
+Rate Interface
+,,,,,,,,,,,,,,
+
+The Rate interface can be implemented by two messages.
+
+The messages are ``mav_msgs/AttitudeThrust`` [#attitudethrust]_ and ``mav_msgs/RollPitchYawrateThrust`` [#rpythrust]_.
+These are both well established messages which have evolved from usage.
+They are well established and have several existing implementations.
+The ``mav_msgs/RollPitchYawrateThrust`` is a convenience message to support low level controllers that are not aware of the absolute yaw.
+
+They diverge from the common ``geometry_msgs/Twist`` [#twist]_ used by ground robots.
+A ``Twist`` based inteface could be provided for backwards compatability as well.
+It was chosen do diverge here since this is a much more natural interface for aerial vehicles.
+
+* Datatype: ``mav_msgs/AttitudeThrust`` [#attitudethrust]_
+* Topic: ``command_velocity``
+* Datatype: ``mav_msgs/RollPitchYawrateThrust`` [#rpythrust]_
+* Topic: ``command_velocity_rpy``
+
+Implementations of this interface should accept messages on either topic/datatype.
+Publishers are expected to choose only one.
+
+Acceleration Interface
```

Comment:
See above comment about attitude interface about renaming. Though i think that "rate" is undercommunicating if not between position and acceleration.